pkgs/development/python-modules/apache-airflow/default.nix

```diff
@@ -102,94 +103,136 @@ buildPythonPackage rec {
-      --replace "python-daemon>=2.1.1, <2.2" "python-daemon" \
+      --replace "funcsigs>=1.0.0, <2.0.0" "funcsigs" \
+      --replace "flask-swagger>=0.2.13, <0.3" "flask-swagger" \
+      --replace "python-daemon>=2.1.1" "python-daemon" \
       --replace "alembic>=1.0, <2.0" "alembic" \
       --replace "markdown>=2.5.2, <3.0" "markdown" \
-      --replace "future>=0.16.0, <0.17" "future" \
+      --replace "future>=0.16.0, <0.19" "future" \
       --replace "tenacity==4.12.0" "tenacity" \
-      --replace "text-unidecode==1.2" "text-unidecode" \
       --replace "tzlocal>=1.4,<2.0.0" "tzlocal" \
       --replace "sqlalchemy~=1.3" "sqlalchemy" \
-      --replace "gunicorn>=19.5.0, <20.0" "gunicorn"
+      --replace "sqlalchemy_jsonfield~=0.9" "sqlalchemy_jsonfield>=0.1" \
+      --replace "gunicorn>=19.5.0, <21.0" "gunicorn" \
+      --replace "werkzeug<1.0.0" "werkzeug" \
+      --replace "importlib_resources~=1.4" "importlib_resources" \
+      --replace "importlib-metadata~=2.0" "importlib-metadata>=1.0" \
+      --replace "marshmallow-sqlalchemy>=0.16.1, <0.24.0" "marshmallow-sqlalchemy" \
+      --replace "requests>=2.20.0, <2.24.0" "requests>=2.20.0" \
+      --replace "lazy_object_proxy<1.5.0" "lazy-object-proxy"
+    # NOTE: preference above is to relax the constraint completely, but when the constraint
+    # is of format ``package>1.0.0; python_version<"3.7"``, we have to put ANY bound
 
-    # dumb-init is only needed for CI and Docker, not relevant for NixOS.
-    substituteInPlace setup.py \
-      --replace "'dumb-init>=1.2.2'," ""
+    # fix issues w/ upgrade to pendulum >= 2.0
+    substituteInPlace airflow/settings.py \
+      --replace "from pendulum import Pendulum" "from pendulum import DateTime as Pendulum"
+    substituteInPlace tests/core/test_core.py \
+      --replace "from pendulum import utcnow" "import pendulum; import functools; utcnow = functools.partial(pendulum.now, 'UTC')"
 
-    substituteInPlace tests/core.py \
-      --replace "/bin/bash" "${stdenv.shell}"
+    # --replace "from pendulum import utcnow" "from datetime import datetime as _dt; utcnow = _dt.utcnow"
   '';
 
   # allow for gunicorn processes to have access to python packages
   makeWrapperArgs = [ "--prefix PYTHONPATH : $PYTHONPATH" ];
 
-  checkPhase = ''
-   export HOME=$(mktemp -d)
-   export AIRFLOW_HOME=$HOME
-   export AIRFLOW__CORE__UNIT_TEST_MODE=True
-   export AIRFLOW_DB="$HOME/airflow.db"
-   export PATH=$PATH:$out/bin
+  preCheck = ''
+    export HOME=$(mktemp -d)
+    export AIRFLOW_HOME=$HOME
+    export AIRFLOW__CORE__UNIT_TEST_MODE=True
+    export AIRFLOW_DB="$HOME/airflow.db"
+    export PATH=$PATH:$out/bin
 
-   airflow version
-   airflow initdb
-   airflow resetdb -y
-   nosetests tests.core.CoreTest
-   ## all tests
-   # nosetests --cover-package=airflow
+    airflow version
+    airflow initdb
+    airflow db reset -y
   '';
+  pytestFlagsArray = [
+    # Subset of tests
+    "tests/core/"
+    "--durations=20"
```

Comment:
what is the durations parameter for? can we add a comment about why the number 20 was chosen?

Reply:
It's mostly a testing (as in me testing the derivation before it makes it to nixpkgs): It shows the durations of the 20 slowest tests (i.e. for disabling slow tests). I'm OK taking it out, but it also doesn't hurt to leave it in, it won't fail anything & can be useful if people want to make ``apache-airflow`` build faster in future.